lib/pure/strutils.nim

```diff
@@ -1330,18 +1330,35 @@ proc join*[T: not string](a: openArray[T], sep: string = ""): string {.
     add(result, $x)
 
 type
-  SkipTable = array[char, int]
+  SkipTable* = array[char, int]
 
-{.push profiler: off.}
-proc preprocessSub(sub: string, a: var SkipTable) =
-  var m = len(sub)
-  for i in 0..0xff: a[chr(i)] = m+1
-  for i in 0..m-1: a[sub[i]] = m-i
-{.pop.}
-
-proc findAux(s, sub: string, start, last: int, a: SkipTable): int =
-  # Fast "quick search" algorithm:
-  var
+proc initSkipTable(a: var SkipTable, sub: string)
+  {.noSideEffect, rtl, extern: "nsuInitSkipTable".} =
+  ## Preprocess table `a` for `sub`.
```

Comment:
Could you expand here (or in the sub documentation) on why/when a skip table might be used?

Reply:
@Varriount 
> What happens if a skip table calculated from a different string is given?

Then the function will return the wrong result.

> Could you expand here (or in the sub documentation) on why/when a skip table might be used?

I do not know English very well :(